pkg/cloudevents/transport/nats/transport.go

```diff
@@ -112,7 +112,7 @@ func (t *Transport) HasConverter() bool {
 
 // StartReceiver implements Transport.StartReceiver
 // NOTE: This is a blocking call.
-func (t *Transport) StartReceiver(ctx context.Context) error {
+func (t *Transport) StartReceiver(ctx context.Context) (err error) {
```

Comment:
do not name the return params.

Reply:
This is necessary to  fix your error return problem at the end of the function. You had a comment:
    // TODO: create an error channel to pass this up
A channel is not necessary. A deferred function executes within the scope of the function that declares it, and can set the return values - but only if they are named.

Reply:
Is that right?! Well thanks for that info. Great to know. LGTM

Reply:
Also after a bit of googling I can't find any objections to named returns, and some articles on how they can be helpful: https://golang.org/doc/effective_go.html#named-results

IMO the only danger is the "naked" return, so I don't use it. I always list the return values even if they are the same variables that were named in parameters to avoid shadowing confusion, e.g.

```
func New() (t *Thing, err error) {
  if err := something(); err != nil {
     return // BAD 
     // naked return uses the variables which are still  (nil, nil)
     // The `err :=`  declaration shadows the err result.

     return t, err // GOOD
     // Explicit return evaluated in local context using inner err variable.
     // values are copied to the result variables before doing defers
 }
 ...
}
```